lib/matplotlib/tests/test_legend.py

```diff
@@ -261,6 +262,31 @@ def test_nanscatter():
     ax.grid(True)
 
 
+@image_comparison(baseline_images=['not_covering_scatter'], extensions=['png'])
+def test_not_covering_scatter():
+    colors = ['b','g','r']
+
+    for n in range(3):
+        plt.scatter([n,], [n,], color=colors[n])
+
+    plt.legend(['foo', 'foo', 'foo'], loc='best')
+    plt.gca().set_xlim(-0.5, 2.2)
+    plt.gca().set_ylim(-0.5, 2.2)
+
+
+@image_comparison(baseline_images=['not_covering_scatter_transform'], extensions=['png'])
```

Comment:
Done.

I noticed the rest of the test file violates this :wink: Can be amended in a separat PR.